### ISSUE TITLE
Remove twisted.manhole

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8330]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8330|
|Type|task|
|Created|2016-05-14 09:12:44Z|
|Branch|https://github.com/twisted/twisted/tree/nomanhole-8330|

```
[4:52pm] glyph:
hawkowl: yeah I think you might be in compatibility-exception territory if you want to fix that coverage problem
[4:52pm] hawkowl:
glyph: okay so. like
[4:52pm] glyph:
hawkowl: I would very much like to have something like twisted.manhole again but it might be worse than useless in its current state
```

<details><summary>Searchable metadata</summary>

```
trac-id__8330 8330
type__task task
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__nomanhole_8330 nomanhole-8330
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1463217164874159 1463217164874159
changetime__1474244394264856 1474244394264856
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Happy with this, please review. Builders are green (except twistedchecker which is spurious after removing so many files).

Please see http://twistedmatrix.com/pipermail/twisted-python/2016-May/030325.html, this will need to wait a week (so minimum of the 21st to merge) + needs two more signoffs before a merge, but I see no reason why the review can't happen before then.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Changes looks good... but there is one thing that worries me.


The ZshIntegrationTests contained a reference to twisted.scripts.manhole.MyOptions and the buildbot was still green.. so there is something wrong with that test.

I have pushed a commit with removed references to manhole.

Please check my latest commit on this branch.

Please also check why the test was still green and fix or create a new ticket.

It would be good to have the test fixed here.

Please address the comments and merge.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

In changeset ebc17b6ef8d470f96d246c013755cd6000edde22
```
#!CommitTicketReference repository="" revision="ebc17b6ef8d470f96d246c013755cd6000edde22"
Merge nomanhole-8330: Remove twisted.manhole

Author: hawkowl
Reviewer: adiroiban
Fixes: #8330
Compatibility Exception Signoffs: hawkowl, glyph, adiroiban
```

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_5"></a>|@rodrigc commented|
|-|-|

In changeset 9481baa4d36d4fb462de5115d3693fb993ac36e1
```
#!CommitTicketReference repository="" revision="9481baa4d36d4fb462de5115d3693fb993ac36e1"
Merge pull request #544 from twisted/8830-mail-py3

Author: berdario
Reviewer: rodrigc
Fixes: #8330

Fix some of the imap tests for Python 3
```
